### ISSUE TITLE
Hang at repair screen

### ISSUE THREAD

**Issue by perryhuynh** [perryhuynh/kcauto#264]
*Please make sure you are on the latest version of kcauto, and that there is not already a ticket open for your issue.*

*Please remove sections of this template you do not find relevant when creating your ticket.*

#### Environment details

* kcauto version (version on first line of `CHANGELOG.md`): 6.5.1
* Viewer/browser: Chrome + KC3
* Operating system and/or VM: Xubuntu 16.04

#### Issue variability
Intermittent

#### Issue
Only 2 ships actually met the criteria to be repaired but it tried to look for a third one and hung at the repair screen.

#### Log
```
[2018-09-15 11:32:30] At home
[2018-09-15 11:32:30] Navigating to repair screen.
[log] CLICK on L(227,431)@S(0)[0,0 1920x1080] (529 msec)
[log] CLICK on L(101,572)@S(0)[0,0 1920x1080] (525 msec)
[2018-09-15 11:32:35] Begin repairing fleets.
[2018-09-15 11:32:35] Detecting 1 ships across 1 pages.
[log] CLICK on L(399,598)@S(0)[0,0 1920x1080] (530 msec)
[2018-09-15 11:32:39] Picking damaged ship from combat fleet(s) to repair.
[log] CLICK on L(814,378)@S(0)[0,0 1920x1080] (530 msec)
[log] CLICK on L(1185,544)@S(0)[0,0 1920x1080] (525 msec)
[log] CLICK on L(1044,771)@S(0)[0,0 1920x1080] (532 msec)
[log] CLICK on L(851,730)@S(0)[0,0 1920x1080] (534 msec)
[log] CLICK on L(546,476)@S(0)[0,0 1920x1080] (530 msec)
[2018-09-15 11:32:58] Picking damaged ship from combat fleet(s) to repair.
[log] CLICK on L(867,321)@S(0)[0,0 1920x1080] (532 msec)
[log] CLICK on L(1151,556)@S(0)[0,0 1920x1080] (532 msec)
[log] CLICK on L(1210,797)@S(0)[0,0 1920x1080] (527 msec)
[log] CLICK on L(817,741)@S(0)[0,0 1920x1080] (528 msec)
[log] CLICK on L(388,617)@S(0)[0,0 1920x1080] (535 msec)
[2018-09-15 11:33:13] Picking damaged ship from combat fleet(s) to repair.
```

#### Config
```
RepairLimit: minor
```

**Comment by mrmin123** [comment]
Do you have a portion of the log that actually says how many ships need to be repaired?

**Comment by perryhuynh** [comment]
```
[2018-09-15 11:31:59] No night battle specified for node #3.
[2018-09-15 11:31:59] Declining night battle.
[log] CLICK on L(477,484)@S(0)[0,0 1920x1080] (528 msec)
[2018-09-15 11:32:13] Fleet 1 damage counts: 1 heavy / 1 moderate / 1 minor
[2018-09-15 11:32:23] 1 ship(s) damaged above threshold. Retreating.
[2018-09-15 11:32:23] Retreating from sortie.
```
I don't have a way of getting the results screen but I have at least this from the replay. Only 2 ships met the repair threshold and I did not have anything else in the dock that wasn't repaired.
![image](https://user-images.githubusercontent.com/9361780/45589583-fe167280-b8dc-11e8-978a-fc5a6d558fa7.png)

**Comment by mrmin123** [comment]
So the repair module is correctly trying to repair 3 ships, it's the combat fleet module that incorrectly identified a ship with minor damage. I'll do a check for this in a bit.

**Comment by mrmin123** [comment]
@perryhuynh I couldn't reproduce with this lineup:
![image](https://user-images.githubusercontent.com/4326866/45589807-48015780-b8e1-11e8-9321-d3a807ce3b8c.png)

```
+  Sikuli match object for 'ship_state_dmg_minor.png' in window 'Chrome'
+    with minimum similarity of 0.85:
[log] toggle highlight M[508,470 7x42]@S(S(0)[0,0 1920x1080]) S:1.00 C:511,491 [-1 msec]: true
M[508,470 7x42]@S(S(0)[0,0 1920x1080]) S:1.00 C:511,491 [-1 msec]


+  Sikuli match object for 'ship_state_dmg_moderate.png' in window 'Chrome'
+    with minimum similarity of 0.85:
[log] toggle highlight M[508,674 7x42]@S(S(0)[0,0 1920x1080]) S:1.00 C:511,695 [-1 msec]: true
M[508,674 7x42]@S(S(0)[0,0 1920x1080]) S:1.00 C:511,695 [-1 msec]


+  Sikuli match object for 'ship_state_dmg_heavy.png' in window 'Chrome'
+    with minimum similarity of 0.85:
[log] toggle highlight M[508,538 7x42]@S(S(0)[0,0 1920x1080]) S:1.00 C:511,559 [-1 msec]: true
M[508,538 7x42]@S(S(0)[0,0 1920x1080]) S:1.00 C:511,559 [-1 msec]
[log] toggle highlight M[508,742 7x42]@S(S(0)[0,0 1920x1080]) S:1.00 C:511,763 [-1 msec]: true
M[508,742 7x42]@S(S(0)[0,0 1920x1080]) S:1.00 C:511,763 [-1 msec]
```

**Comment by mrmin123** [comment]
Perhaps the approach I should take is so that the repair module doesn't infinitely loop, even with the incorrect damage count (although that is concerning for its own set of reasons).

**Comment by mrmin123** [comment]
The root cause of this was resolved in 6.6.2.
